packages/gatsby-source-contentful/src/plugin-options.js

```diff
@@ -24,40 +25,152 @@ const createPluginConfig = pluginOptions => {
   }
 }
 
-const optionsSchema = Joi.object().keys({
-  accessToken: Joi.string().required().empty(),
-  spaceId: Joi.string().required().empty(),
-  host: Joi.string().empty(),
-  environment: Joi.string().empty(),
-  downloadLocal: Joi.boolean(),
-  localeFilter: Joi.func(),
-  forceFullSync: Joi.boolean(),
-  pageLimit: Joi.number().integer(),
-  proxy: Joi.object().keys({
-    host: Joi.string().required(),
-    port: Joi.number().required(),
-    auth: Joi.object().keys({
-      username: Joi.string(),
-      password: Joi.string(),
-    }),
-  }),
-  useNameForId: Joi.boolean(),
-  // default plugins passed by gatsby
-  plugins: Joi.array(),
-  richText: Joi.object()
-    .keys({
-      resolveFieldLocales: Joi.boolean(),
-    })
-    .default({}),
+const Joi = joi.extend({
+  type: /^s/,
```

Comment:
What does this syntax mean?

Reply:
This syntax means that we extend _all_ keys with the `.dotenv` method!

Taken from https://github.com/sideway/joi/commit/03adf22eb1f06c47d1583617093edee3a96b3873, will add a comment explaining that and linking to this!